src/s3/document/s3DocumentProvider.ts

```diff
@@ -10,9 +10,7 @@ import { getLogger } from '../../shared/logger/logger'
 import { showOutputMessage } from '../../shared/utilities/messages'
 
 export class S3DocumentProvider implements vscode.TextDocumentContentProvider {
-    public constructor() {}
-
-    onDidChange?: vscode.Event<vscode.Uri> | undefined
+    public constructor(public onDidChange: vscode.Event<vscode.Uri>) {}
```

Comment:
Are events necessary here? Events are great for high decoupling but also restrict communication to only what is described by the event. I don't see a need for such high decoupling currently, though I could imagine scenarios where it might be desired.

Reply:
Do you mean like the assignment inside the constructor's parameters?

Reply:
No I mean like we're using this event to communicate `FileViewerManager` ----> `S3DocumentProvider` right?
Well couldn't the manager just be created with the document provider instance and then call a method directly? Events make sense when we don't know who will be listening, but in this case it's pretty clear.

Reply:
I am not very sure, honestly I am still a little confused on what we did here. Maybe @bryceitoc9 can explain better this decision.

Reply:
Event is necessary to trigger VS Code to update the document. I think the justification for an event instead of the manager calling this to emit an event was a circular dependency concern.

Reply:
Okay that makes sense. Yes I could see the potential for circular dependency here.